core/src/loader.rs

```diff
@@ -206,10 +203,7 @@ pub fn load_token_account(
 /// # Returns
 /// * `Result<(), ProgramError>` - The result of the operation
 pub fn load_token_mint(info: &AccountInfo) -> Result<(), ProgramError> {
-    if !(info.owner.eq(&spl_token::id()) || info.owner.eq(&spl_token_2022::id())) {
```

Comment:
should we just check for spl token program here?

Reply:
the description talks about removing Token2022, imo should keep the spl_token::id check here and remove the spl22 check

Reply:
agreed and reverted